internal/jimm/juju/cache.go

```diff
@@ -57,9 +61,26 @@ func (d *cacheDialer) Dial(ctx context.Context, ctl *dbmodel.Controller, mt name
 	}
 }
 
-func (d *cacheDialer) dial(ctx context.Context, ctl *dbmodel.Controller, requiredPermissions map[string]string) (interface{}, error) {
+func mapKey(user *openfga.User, ctl *dbmodel.Controller, requiredPermissions map[string]string) string {
```

Comment:
This needs a better name and some docstrings. It looks like we're creating a cached dialer per user per controller?

Reply:
that's exactly right. Since we now dial as user, we need to cache controller connection per controller, user and the set of permissions. that is why i disabled connection caching in this PR until we're absolutely sure we need it.

Reply:
Can we still give this function a more descriptive name even though we're not currently using the cache.